### DISCUSSION TITLE
GraphQL support in Starlette is deprecated

### DISCUSSION THREAD

**Discussion by memark** [fastapi/fastapi#6551] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
n/a
```

### Description

>GraphQL support in Starlette is deprecated as of version 0.15 and will be removed in a future release.
>It is also incompatible with Python 3.10+. 

https://www.starlette.io/graphql/

Is there a plan for moving away from Starlette? Maybe using one of the alternatives suggested on that page?

### Operating System

macOS

### FastAPI Version

0.68.0

### Python Version

3.9.5

**Comment by Honda-a** [comment]
is there any update on this ?
I'm planning to use fastapi with graphql and python3.10 but if there is no fix for this I may need to change my FW

**Comment by Honda-a** [comment]
@tiangolo 
I have some free time and can work on this if you this is of any priority let's discuss about the details

**Comment by Honda-a** [comment]
Okay I found this we can integrate strawberry for graphql
https://strawberry.rocks/docs/integrations/fastapi

**Comment by Kludex** [comment, marked as answer]
There's no plan to move GraphQL to FastAPI. FastAPI is going to follow Starlette's decision.

We are probably going to recommend Strawberry at some point as an alternative, mostly because they use Pydantic and are "autocomplete oriented" as FastAPI.
